src/Agent.Plugins/PipelineCache/SavePipelineCacheV0.cs

```diff
@@ -63,6 +63,12 @@ public override async Task RunAsync(AgentTaskPluginExecutionContext context, Can
             CancellationToken token)
         {
             string contentFormatValue  = context.Variables.GetValueOrDefault(ContentFormatVariableName)?.Value ?? string.Empty;
+            string calculatedFingerPrint = context.TaskVariables.GetValueOrDefault(CalculatedFingerPrintVariableName)?.Value ?? string.Empty;
+
+            if(!string.IsNullOrWhiteSpace(calculatedFingerPrint) && !fingerprint.ToString().Equals(calculatedFingerPrint, StringComparison.Ordinal))
+            {
+                context.Warning($"Fingerprint has been modified; original fp: {calculatedFingerPrint}, modified fingerprint {fingerprint}");
```

Comment:
line them up for easier comparison
```
Fingerprint has been modified;\n
original fingerprint: {}\n
modified fingerprint: {}\n

Reply:
Fingerprint isn't modified hmm.   "The given cache key has changed in its resolved value between restore and save steps."?

Reply:
I think fingerprint is internal only

Reply:
Yup - modified the message.